src/gatsby/src/components/search/noSearchResults.js

```diff
@@ -2,93 +2,133 @@ import React from "react"
 import { Link } from "gatsby"
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
 import { faSearch, faArrowUpRightFromSquare } from "@fortawesome/free-solid-svg-icons"
+import PropTypes from "prop-types"
 
 import "../../styles/search.scss"
 
+const MapLink = () => {
+  return (
+    <a
+      className="map-link"
+      href="https://governmentofbc.maps.arcgis.com/apps/webappviewer/index.html?id=077ef73a1eae4ca88f2bafbb831215af&query=British_Columbia_Parks_Ecological_Reserves_and_Protected_Areas_8747,ORCS_PRIMARY,0000"
+    >
+      Map of parks
+      <FontAwesomeIcon icon={faArrowUpRightFromSquare} className="link-icon" />
+    </a>
+  )
+}
+
 const NoSearchResults = ({
-  hasCity,
-  hasPark,
-  hasFilter,
   handleClickClearCity,
-  handleKeyDownClearCity,
   handleClickClearPark,
-  handleKeyDownClearPark,
   handleClickClearFilter,
-  handleKeyDownClearFilter,
+  page,
 }) => {
+  const renderNoResultsMessage = () => {
+    switch (page) {
+      case "find-a-park":
+        return "No parks match your search."
+      case "park-operating-dates":
+      case "a-z-list":
+        return "No parks match your selection."
+      case "approved":
+        return "No documents match your selection."
+      default:
+        return ""
+    }
+  }
+  // event handlers
+  const handleClearAll = () => {
+    handleClickClearCity()
+    handleClickClearPark()
+    handleClickClearFilter()
+  }
+  const handleKeyDownClearAll = e => {
+    if (e.key === "Enter" || e.key === " ") {
+      e.preventDefault()
+      handleClearAll()
+    }
+  }
+
   return (
     <div className="no-results-container">
       <div className="no-results-header">
         <FontAwesomeIcon icon={faSearch} className="search-icon" />
         <div className="no-results-header--right">
           <h2>Sorry...</h2>
-          <p><b>no parks matched your search.</b></p>
+          <p><b>{renderNoResultsMessage()}</b></p>
         </div>
       </div>
-      <p><b>You could:</b></p>
-      <ul>
-        {(hasCity || hasPark) && (
-          <li>Change your search term</li>
-        )}
-        {hasCity && (
-          <li>
-            <button
-              tabIndex={0}
-              className="btn btn-link"
-              aria-label="Remove city name"
-              onClick={handleClickClearCity}
-              onKeyDown={handleKeyDownClearCity}
+      {page === "find-a-park" && (
+        <>
+          <p><b>You could:</b></p>
+          <ul>
+            <li>Try a different search term</li>
+            <li>
+              <button
```

Comment:
You can "press" a button with a mouse click, or by focusing it and pressing space (or enter) but I _think_ all 3 of these event swill fire a `click` event, so you don't need to specifically handle onKeyDown. (Double check if that's true in React, though...)

Reply:
Good catch! Yes it works in React as well.